cmd/ipfs/daemon.go

```diff
@@ -629,19 +629,50 @@ func mountFuse(req *cmds.Request, cctx *oldcmds.Context) error {
 }
 
 func maybeRunGC(req *cmds.Request, node *core.IpfsNode) (<-chan error, error) {
-	enableGC, _ := req.Options[enableGCKwd].(bool)
-	if !enableGC {
+	gc, err := initGC(req, node)
+	if err != nil {
+		return nil, err
+	}
+	if !gc.EnableGC {
+		log.Warningf("GC: %s", "Repo GC is disabled.")
 		return nil, nil
 	}
 
 	errc := make(chan error)
-	go func() {
-		errc <- corerepo.PeriodicGC(req.Context, node)
-		close(errc)
-	}()
+	// if duration larger than 0, it means GC is enabled.
+	if int64(gc.GCPeriod) > 0 {
+		go func() {
+			errc <- corerepo.PeriodicGC(req.Context, gc)
+			close(errc)
+		}()
+	}
+
 	return errc, nil
 }
 
+func initGC(req *cmds.Request, node *core.IpfsNode) (*corerepo.GC, error) {
+	enableGC, _ := req.Options[enableGCKwd].(bool)
+	cfg, err := node.Repo.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.Datastore.GCPeriod == "" {
+		cfg.Datastore.GCPeriod = "1h"
+	}
+
+	gc, err := corerepo.NewGC(node)
+	if err != nil {
+		return nil, err
+	}
+	gc.EnableGC = enableGC
```

Comment:
Let's keep `enableGC` set-value and load-value operations as close as possible, move the `enableGC, _ := req.Options[enableGCKwd].(bool)` line down here if it's not used anywhere else.

Reply:
Because i should get this param in other function.
```
func maybeRunGC(req *cmds.Request, node *core.IpfsNode) (<-chan error, error) {
	gc, err := initGC(req, node)
	if err != nil {
		return nil, err
	}
	if !gc.EnableGC {
		log.Warningf("GC: %s", "Repo GC is disabled.")
		return nil, nil
	}
.........
```